driver-sync/src/test/functional/com/mongodb/client/AbstractTransactionsTest.java

```diff
@@ -344,6 +244,128 @@ public void shouldPassAllOutcomes() {
         }
     }
 
+    private void executeOperations(final BsonArray operations, final boolean throwExceptions) {
```

Comment:
Extracted this as method so that it can be called recursively from withTransaction closure below.